symengine/printer.cpp

```diff
@@ -181,6 +181,16 @@ void StrPrinter::bvisit(const EmptySet &x)
     str_ = "EmptySet";
 }
 
+void StrPrinter::bvisit(const Union &x)
+{
+    std::ostringstream s;
+    s << **x.container_.begin();
+    for (auto it = ++(x.container_.begin()); it != x.container_.end(); ++it) {
+        s << " U " << **it;
```

Comment:
Don't use `<< **it;` as it will prevent extending the priitnter. Use ` << apply(*it);`